### DISCUSSION TITLE
Looking for non-deprecated package with hast Element and Properties types

### DISCUSSION THREAD

**Discussion by codepunkt** [unifiedjs/unified#212] [General] [answered]
I'm using `import type { Element, Properties } from 'hast'` in a remark plugin of mine.

However, on installation of `hast` or `@types/hast` I get this deprecation warning:
![image](https://user-images.githubusercontent.com/603683/212567689-1b338d65-64c8-4450-b784-b837f07110cb.png)

What other package in the unified ecosystem can I read these types from?

**Comment by ChristianMurphy** [comment, marked as answer]
HAST is a types only project.
`@types/hast` is correct if you need the typings.

There used to be an implementation with the name `hast` as well, that has been replaced with `rehype`.
Don't install `hast`  use `rehype` as a parser.
